### PR TITLE
Feature #7340. Add the itemStyle to legend

### PR DESCRIPTION
**Pull request by yufeng04**
Add the itemStyle to legend symbol.
* The default borderWidth of legend symbol is 0, so the borderColor only can be seen when the legend.itemStyle.borderWidth isn't 0.
*  `legend.itemStyle.borderWidth !== 0`
    1. the borderWidth of legend symbol only depends on  `legend.itemStyle.borderWidth`
    2. the boderColor of legend symbol depends on `legend.itemStyle.borderColor` and `series[i].itemStyle.borderColor`, and the `legend.itemStyle.borderColor` has higher priority.
    3. The borderColor of `emptyCircle` legend symbol is the same as color of series
    4. The legend borderColor is incompatible with `barBorderColor`
* [The former PR](https://github.com/apache/incubator-echarts/pull/11295)

### CODE REVIEW

File: src/visual/seriesColor.js [hunk 1]
```diff
@@ -48,9 +52,13 @@ export default {
...
                 var itemModel = data.getItemModel(idx);
                 var color = itemModel.get(colorAccessPath, true);
+                var borderColor = itemModel.get(borderColorAccessPath, true);
                 if (color != null) {
                     data.setItemVisual(idx, 'color', color);
                 }
+                if (borderColor != null) {
+                    data.setItemVisual(idx, 'borderColor', borderColor);
+                }
             };
```
Comment (100pah) [inline src/visual/seriesColor.js]: image

Why they are different

File: src/visual/seriesColor.js [hunk 2]
```diff
@@ -48,9 +52,13 @@ export default {
...
 
             return { dataEach: data.hasItemOption ? dataEach : null };
```
Comment (100pah) [inline src/visual/seriesColor.js]: image

If use `scatter`:
If `symbol: emptyCircle`: The color of scatter is different with legend.
If `symbol: circle` `border` set, legend do not show the border.

```js



var option4 = {
    tooltip: {
        trigger: 'axis',
        axisPointer: {
            type: 'shadow'
        }
    },
    legend: {
        // itemStyle: {
        //     borderWidth: 3
        // },
        // selected: {
        //     '联盟广告': false
        // }
    },
    grid: {
        left: '3%',
        right: '4%',
        bottom: '3%',
        containLabel: true
    },
    xAxis: [
        {
            type: 'category',
            data: ['周一', '周二', '周三', '周四', '周五', '周六', '周日']
        }
    ],
    yAxis: [
        {
            type: 'value'
        }
    ],
    series: [
        {
            name: '直接访问',
            type: 'scatter',
            data: [320, 332, 301, 334, 390, 330, 320],
            symbol: 'emptyCircle',
            itemStyle: {
                borderColor: '#5e1',
                borderWidth: 3
            }
        },
        {
            name: '邮件营销',
            type: 'scatter',
            data: [120, 132, 101, 134, 90, 230, 210],
            symbol: 'emptyCircle',
            itemStyle: {
                borderColor: '#ae2',
                borderWidth: 5
            }
        },
        {
            name: '邮件营销2',
            type: 'scatter',
            data: [21, 33, 11, 22, 22, 11, 22],
            symbol: 'circle',
            itemStyle: {
                borderColor: '#912',
                borderWidth: 5
            }
        }
    ]
};
// chart1.setOption(option4);
testHelper.create(echarts, 'plain4', {
    title: [
        '(Legend symbol of line)??',
    ],
    option: option4,
    height: 300
});

```

File: src/visual/dataColor.js [hunk 3]
```diff
@@ -74,6 +76,23 @@ export default function (seriesType) {
                     // Set data all color for legend
                     dataAll.setItemVisual(rawIdx, 'color', singleDataColor);
                 }
+
+                if (!singleDataBorderColor) {
+                    // FIXME Performance
+                    var itemModel = dataAll.getItemModel(rawIdx);
```
[the remark this answers is not in the document]
Comment (yufeng04) [reply src/visual/dataColor.js]: Thanks！I didn't take this into account.

File: src/visual/seriesColor.js (hunk 1 above)
Comment (yufeng04) [reply src/visual/seriesColor.js]: The emptyCircle symbol doesn't have fillColor, so the strokeColor should be the same as color of corresponding series

**Pull request merged**